### PR TITLE
hw6: add routing and requests to api, minor changes in api

### PR DESCRIPTION
**Pull request by ValeriyPetryna**

### CODE REVIEW

**Comment by ValeriyPetryna** [comment]
image

File: src/app/post/components/post-page/post-page.component.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        error: (e) => {
+          console.error(e);
+          this.notFound = true;
+        }
+      })
+    } else {
+      this.notFound = true;
+    }
+
+    // should get posts on init, subscribe on service data or use async pipe
```
Comment (MMaksymivUa) [inline src/app/post/components/post-page/post-page.component.ts:35]: please add prettier/other formatter on save

File: src/app/post/components/post-page/post-page.component.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+})
+export class PostPageComponent implements OnInit, OnDestroy {
+  public id!: string | null;
+  public notFound!: boolean;
+  public post!: Post;
+
+  constructor(private blogService: BlogService, private route: ActivatedRoute, private http: HttpClient) {}
+
+  ngOnInit(): void {
+    this.id = this.route.snapshot.paramMap.get('id');
```
Comment (MMaksymivUa) [inline src/app/post/components/post-page/post-page.component.ts:22]: Do we have a possibility to navigate from one post-page to another? For example: we are one post with id=1, inside this route we have a link to post with id=2

File: src/app/shared/services/blog.service.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+})
+export class BlogService {
+  public postStream = new BehaviorSubject<Post[]>([]);
+
+  constructor (public http: HttpClient) {}
+
+  public feedPosts(): BehaviorSubject<Post[]> {
+    this.http.get<Post[]>(basePostsURL).subscribe(this.postStream);
+    return this.postStream;
+  }
```
Comment (MMaksymivUa) [inline src/app/shared/services/blog.service.ts:19]: ``return this.http.get<Post[]>(basePostsURL)``

File: src/app/shared/services/blog.service.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  };
+
+  public getPostData(): Observable<Post[]> {
+    return this.postStream.asObservable();
+  }
+
+  public updatePostData(post: Post): void {
+    this.postStream.next([...this.postStream.getValue(), post]);
+  }
+}
```
Comment (MMaksymivUa) [inline src/app/shared/services/blog.service.ts:48]: no HTTP headers?

File: src/app/post/components/post-page/post-page.component.ts (hunk 1 above)
Comment (ValeriyPetryna) [reply src/app/post/components/post-page/post-page.component.ts:35]: added prettier, thanks

File: src/app/post/components/post-page/post-page.component.ts (hunk 2 above)
Comment (ValeriyPetryna) [reply src/app/post/components/post-page/post-page.component.ts:22]: for now, I don't have such a possibility, maybe in the future, we will have a possibility to jump to another current author post or something like this.

File: src/app/shared/services/blog.service.ts (hunk 3 above)
Comment (ValeriyPetryna) [reply src/app/shared/services/blog.service.ts:19]: Thanks, it was changed in the next PR

File: src/app/shared/services/blog.service.ts (hunk 4 above)
Comment (ValeriyPetryna) [reply src/app/shared/services/blog.service.ts:48]: Thanks for your advice, I will add HTTP headers with authorization and authentication.

**Pull request merged**